src/com/owncloud/android/files/services/TransferRequester.java

```diff
@@ -126,7 +126,14 @@ private void uploadsUpdate(Context context, Account account, OCFile[] existingFi
         intent.putExtra(FileUploader.KEY_LOCAL_BEHAVIOUR, behaviour);
         intent.putExtra(FileUploader.KEY_FORCE_OVERWRITE, forceOverwrite);
 
-        context.startService(intent);
+        // Since in Android O and above the apps in background are not allowed to start background
+        // services and available offline feature may try to do it, this is the way to proceed
+        if (context instanceof AvailableOfflineSyncJobService && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
it's quite strange to change this logic based on the type f the context, especially when it was just used to start a service. Take into account that the Android context may be passed in very different ways and   Looks like an architecture smell. Probably overloading the uploadsUpdate method with a new flag and passing it around would help. Just to consider, not sure if feasible.